src/stepFunctions/commands/visualizeStateMachine/aslVisualizationCDKManager.ts

```diff
@@ -0,0 +1,108 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+
+import { AslVisualizationCDK } from './aslVisualizationCDK'
+import { AbstractAslVisualizationManager } from './abstractAslVisualizationManager'
+import { ConstructNode } from '../../../cdk/explorer/nodes/constructNode'
+import { getLogger } from '../../../shared/logger'
+import { normalize } from '../../../shared/utilities/pathUtils'
+
+export class AslVisualizationCDKManager extends AbstractAslVisualizationManager {
+    protected readonly name: string = 'AslVisualizationCDKManager'
+    protected readonly managedVisualizations: Map<string, Map<string, AslVisualizationCDK>> = new Map<
+        string,
+        Map<string, AslVisualizationCDK>
+    >()
+
+    public constructor(extensionContext: vscode.ExtensionContext) {
+        super(extensionContext)
+    }
+
+    //vscode.Memento passed in to update StateMachineGraphCache on each visualization
+    public async visualizeStateMachine(
+        globalStorage: vscode.Memento,
+        node: ConstructNode
+    ): Promise<vscode.WebviewPanel | undefined> {
+        const logger = getLogger()
+
+        const cdkOutPath = node.id.replace(`/tree.json/${node.tooltip}`, ``)
+        const workspaceName = this.getCDKAppWorkspaceName(cdkOutPath)
+
+        const existingVisualization = this.getExistingVisualization(workspaceName, node.tooltip)
+
+        if (existingVisualization) {
+            existingVisualization.showPanel()
+
+            return existingVisualization.getPanel()
+        }
+
+        const stateMachineName = node.label
+        const appName = node.tooltip.replace(`/${stateMachineName}`, ``)
+        const templatePath = normalize(`${cdkOutPath}/${appName}.template.json`)
+        const uri = vscode.Uri.file(templatePath)
+
+        // Existing visualization does not exist, construct new visualization
+        try {
+            await this.cache.updateCache(globalStorage)
+
+            const textDocument = await vscode.workspace.openTextDocument(uri)
+            const newVisualization = new AslVisualizationCDK(textDocument, templatePath, appName, stateMachineName)
+            if (newVisualization) {
+                this.handleNewVisualization(workspaceName, newVisualization)
+                return newVisualization.getPanel()
+            }
+        } catch (err) {
+            this.handleErr(err as Error, logger)
+        }
+
+        return
```

Comment:
Unnecessary return (nit), but it does beg the question under what cases you'll return undefined? Is it only in the error cases, or are there cases where you'll not hit the `else` in the `try` block? In these cases, should there be some sort of user feedback?

Reply:
`existingVisualization.getPanel()` returns undefined if the panel has already been disposed. Removed the unnecessary return and the else block as the `newVisualization` should only be undefined if there is some error thrown during the constructor of `AslVisualizationCDK`, which gets handled in the catch.